docs/architecture/notes.md

```diff
@@ -196,10 +195,34 @@ hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)
 !!! info
     To compute a note's hash, we do not need to know the note's `serial_num`. Knowing the hash of the `serial_num` (as well as `script_hash`, `input_hash` and `note_vault`) is also sufficient. We compute the hash of `serial_num` as `hash(serial_num, [0; 4])` to simplify processing within the VM._
 
-## Note discovery
+## Note discovery (Note Tags)
+
+Note discovery describes the process of Miden clients finding notes they want to consume. Miden clients can query the Miden node for notes carrying a certain note tag. Note tag`s are best effort filters for notes registered with the network. They are light-weight values used to speed up queries. Clients can follow tags of a certain use case, e.g. swap scripts, or agree upon a group of users on certain tags for their specific use case. Tags are also used by the operator to identify notes intended for network execution including the corresponding information on how to execute.
```

Comment:
```suggestion
Note discovery describes the process of Miden clients finding notes they want to consume. Miden clients can query the Miden node for notes carrying a certain note tag. Note tag`s are best effort filters for notes registered on the network. They are light-weight values used to speed up queries. Clients can follow tags for a certain use case, e.g. swap scripts, or agree upon a group of users of certain tags for their specific use case. Tags are also used by the operator to identify notes intended for network execution and include the corresponding information on how to execute.
```

Reply:
I'm not sure what this means "or agree upon a group of users of certain tags for their specific use case"...

Reply:
I change it to 

```
Clients can follow tags for a certain use case, e.g. swap scripts, or agree with a group of users on certain tags for their specific use case.
```